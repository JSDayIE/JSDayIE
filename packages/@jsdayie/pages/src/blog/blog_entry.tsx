import React from "react";
import { MarkDown, Container } from "@jsdayie/components";

export interface BlogEntryProps {
  markdown: string;
}

export const BlogEntry: React.FC<BlogEntryProps> = (props) => {
  const { markdown } = props;
  return (
    <Container>
      <MarkDown markdown={markdown} />
    </Container>
  );
};
