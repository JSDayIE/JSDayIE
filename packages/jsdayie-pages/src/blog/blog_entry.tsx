import React from "react";
import { MarkDown, Container } from "@jsdayie/components";

interface BlogEntryProps {
  markdown: string;
}

export const BlogEntry: React.FC<BlogEntryProps> = props => {
  return (
    <Container>
      <MarkDown markdown={props.markdown} />
    </Container>
  );
}