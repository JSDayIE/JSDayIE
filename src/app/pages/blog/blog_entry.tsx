import React from "react";
import * as showdown from "showdown";

interface MarkDownProps {
  text: string;
}

function MarkDown(props: MarkDownProps) {
  const converter = new showdown.Converter();
  const html = converter.makeHtml(props.text);
  return <div>{html}</div>;
}

export class BlogEntry extends React.Component {
  render() {
    return (
        <div>TODO</div>
    );
  }
}
