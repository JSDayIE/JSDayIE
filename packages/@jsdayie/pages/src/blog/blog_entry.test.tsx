import React from "react";
import renderer from "react-test-renderer";
import { BlogEntry } from "./blog_entry";

test("BlogEntry", async () => {
  const component = renderer.create(<BlogEntry markdown="**test**" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
