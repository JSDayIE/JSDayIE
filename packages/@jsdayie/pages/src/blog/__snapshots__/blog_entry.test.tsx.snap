// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogEntry 1`] = `
<div
  className="container content"
>
  <div
    className="markdown"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p><strong>test</strong></p>",
        }
      }
    />
  </div>
</div>
`;
